easybuild/easyblocks/d/doris.py

```diff
@@ -73,17 +73,63 @@ def configure_step(self):
 
         run_cmd_qa('./configure', qa, std_qa=std_qa, log_all=True, simple=True)
 
+    def build_step(self):
+        """Custom build procedure for Doris."""
+        common_buildopts = self.cfg['buildopts']
+
+        # build Doris
+        change_dir(os.path.join(self.cfg['start_dir'], 'src'))
+
         # override some of the settings via options to 'make'
-        lflags = "-L%s -lfftw3 " % os.path.join(fftw, 'lib')
+        lflags = "-L%s -lfftw3 " % os.path.join(get_software_root('FFTW'), 'lib')
         lflags += "-L%s %s" % (os.getenv('LAPACK_LIB_DIR'), os.getenv('LIBLAPACK_MT'))
         self.cfg.update('buildopts', 'LFLAGS="%s"' % lflags)
-
         self.cfg.update('buildopts', 'CFLAGSOPT="%s \$(DEFS)"' % os.getenv('CXXFLAGS'))
 
+        super(EB_Doris, self).build_step()
+
+        # build SARtools
+        change_dir(os.path.join(self.cfg['start_dir'], 'SARtools'))
+
+        self.cfg['buildopts'] = common_buildopts
+        self.cfg.update('buildopts', 'CC="%s"' % os.getenv('CXX'))
+        cflags = os.getenv('CFLAGS') + " -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE"
+        self.cfg.update('buildopts', 'CFLAGS="%s"' % cflags)
```

Comment:
well, I need to define `CFLAGS`, but I should use the `$CXXFLAGS` value as defined by EasyBuild here, yes